src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Refresh.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        protected override bool RequiresPagedDataForParamCore(TexlNode[] args, int paramIndex, TexlBinding binding)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.Assert(paramIndex >= 0 && paramIndex < args.Length);
+            Contracts.AssertValue(binding);
+            Contracts.Assert(binding.IsPageable(args[paramIndex].VerifyValue()));
+
+            // Refresh only needs metadata. No actual data from datasource is required.
+            return false;
+        }
+
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
+        {
+            Contracts.AssertValue(binding);
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.AssertAllValid(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            if (!argTypes[0].IsAggregate)
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedAgg);
+                return;
+            }
+
+            // Only certain types of data sources are refreshable.
+            // Note that we're only adding errors we know for sure that the argument refers to a non-refreshable
+            // data source. Currently this can only be done for simple FirstNameNodes. For more complex expressions such as:
+            //      Refresh(If(a < 1, Data1, [1,2,3]))
+            // ...we have no way of inferring statically what the possible result of evaluating the expression will be,
+            // so it's not safe to flag the rule with errors. The runtime safety mechanisms will later kick in and prevent
+            // crashes when the rule gets evaluated. If the argument evaluates to a non-refreshable data source, such as
+            // a collection or excel table, the Refresh invocation will simply be a no-op in that case.                        
+            if (binding.TryCastToFirstName(args[0], out FirstNameInfo firstNameInfo) && binding.IsInfoKindDataSource(firstNameInfo) &&
+                firstNameInfo.Data is IRefreshable)
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrDataSourceCannotBeRefreshed);
+            }
+        }
+
+        public override IEnumerable<Identifier> GetIdentifierOfModifiedValue(TexlNode[] args, out TexlNode identifierNode)
+        {
+            Contracts.AssertValue(args);
+
+            identifierNode = null;
+            if (args.Length == 0)
+            {
+                return null;
+            }
```

Comment:
All this code is coming from PA Client and this part is the same (aside the curly brackets)

Even if this part is 100% useless, I trust you on that, I always like defensive code as line 110 would throw in case of a 0-length array. 

I can remove it if you think that's really dead code and we'll never ever be in that situation.

For ref: [https://msazure.visualstudio.com/OneAgile/_git/PowerApps-Client?path=/src/Cloud/DocumentServer.Core/Document/Texl/Builtins/Refresh.cs&version=GBmaster&line=107&lineEnd=108&lineStartColumn=1&lineEndColumn=29&lineStyle=plain&_a=contents](https://msazure.visualstudio.com/OneAgile/_git/PowerApps-Client?path=/src/Cloud/DocumentServer.Core/Document/Texl/Builtins/Refresh.cs&version=GBmaster&line=107&lineEnd=108&lineStartColumn=1&lineEndColumn=29&lineStyle=plain&_a=contents)